### PR TITLE
build with travis and doctr

### PR DESCRIPTION
**Pull request by ocefpaf**
@emiliom I sent a request to activate Travis-CI here, not sure who will get the e-mail.

We'll need this to publish the older versions of the site, a draft and current version.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+language: minimal
+
+sudo: false
+
+env:
+  global:
+    - secure:
```
Comment (ocefpaf) [inline .travis.yml]: Don't merge this yet. I need to get Travis-CI configure first and then use `doctr` to create the encrypted key.

**Comment by leewujung** [comment]
@ocefpaf turns out that I got the email, not sure why.
Should I select All repositories or Only select repositories in the options?

**Comment by amarburg** [comment]
@ocefpaf I also got the email and OK'ed it ... didn't want to hold up the process.

**Comment by ocefpaf** [comment]
Thanks @amarburg and @leewujung! Somehow I'm still not seeing this repo on Travis-CI. Can you check if you have the proper rights to activate it:

https://travis-ci.org/github/oceanhackweek/oceanhackweek.github.io

**Comment by amarburg** [comment]
Looks like it's building now ...

[Screenshot...](https://www.dropbox.com/s/cwonilf130e5c6p/TravisCI_screenshot.jpg?dl=0)

Permissions problem?

**Comment by ocefpaf** [comment]
> Looks like it's building now ...

Yep. Thanks!

**Comment by ocefpaf** [comment]
@emiliom I'm merging this so we can perform a few more tests of this new setup. If things fail we should revert this PR.

**Comment by ocefpaf** [comment]
Adjusting it to https://drdoctr.github.io/recipes.html#using-doctr-with-github-io-pages b/c the `.io` pages are a special case.

@emiliom this will increase the complexity here a little bit b/c our new working branch will be `source` rather than master.

**Comment by emiliom** [comment]
> this will increase the complexity here a little bit b/c our new working branch will be source rather than master.

Ok. But once we merge into master, the same auto-build will happen as it does with master right now, right?

**Comment by emiliom** [comment]
Sorry, I meant merge into source!

**Comment by ocefpaf** [comment]
> Ok. But once we merge into master, the same auto-build will happen as it does with master right now, right?

Yep. `doctr` manages the HTML in the publication branch (`master` in this case) and the HTML generation in another (`source`). The `master` -> `gh-pages` is easier to setup but I'm having some issues here, specially b/c this is a `.io` site, which behaves differently (thanks GitHub!).

If I cannot make this work until tomorrow afternoon I'll revert the changes and look for an alternative to doctor.

**Comment by ocefpaf** [comment]
BTW, the site is down b/c somehow doctr cannot find new modifications even though  there are some. I'm in touch with upstream in https://github.com/drdoctr/doctr/issues/313#issuecomment-606303835, it must be something simple I'm missing.

**Comment by ocefpaf** [comment]
Alright, the site is back up! The main one is at https://oceanhackweek.github.io/ and the 2019 is published on https://oceanhackweek.github.io/ohw19/

We can freeze sites by creating a tag and they will get published are `https://oceanhackweek.github.io/<TAGNAME>`.

I'm working on a way to publish a branch, say draft, to a https://oceanhackweek.github.io/draft, to make it easier to stage modifications.

**Comment by emiliom** [comment]
Wonderful!! I love the idea of using a tag. It really is a "release", so it makes perfect sense. Thank you!!

However, the content of the ohw19 site is mostly wrong at this time. It's the content from the current site (master branch), except for what's coming from the `_config.yml` file, which is correct (ohw19). Also the pages available from the menu are correct.

**Comment by ocefpaf** [comment]
> However, the content of the ohw19 site is mostly wrong at this time. It's the content from the current site (master branch), except for what's coming from the `_config.yml` file, which is correct (ohw19). Also the pages available from the menu are correct.

Yep. There is some fine tweaking that I need to do. Right now I'm puzzled by the fact that the tag overwrote the root site :-/

**Comment by emiliom** [comment]
> Right now I'm puzzled by the fact that the tag overwrote the root site :-/

Right! I hadn't noticed that. The main site and the ohw19 site are identical.

Well, the main thing is that the main site is back online and

**Comment by ocefpaf** [comment]
> Right! I hadn't noticed that. The main site and the ohw19 site are identical.

It should be fixed now. The last missing piece is to change the 2019 site to prepend the `ohw19` slug so the links work as expected.

**Pull request merged**